lib/json/encoder.ex

```diff
@@ -144,18 +144,17 @@ defimpl JSON.Encoder, for: Map do
   def typeof(_), do: :object
 end
 
-#TODO: maybe this should return the result of "inspect" ?
 defimpl JSON.Encoder, for: Any do
-  @any_encode "[Elixir.Any]"
-
   def encode(%{} = struct) do
-    JSON.Encoder.Helpers.dict_encode(Map.to_list(struct))
+    struct
+    |> Map.to_list()
+    |> JSON.Encoder.Helpers.dict_encode()
   end
 
-  def encode(_), do: JSON.Encoder.encode(@any_encode)
+  def encode(x), do: JSON.Encoder.encode("#{inspect x}")
```

Comment:
[`Kernel.inspect/2`](http://elixir-lang.org/docs/stable/elixir/Kernel.html#inspect/2) will return a `String.t`, is there a reason you're wrapping it in string interpolation?